packages/flutter_tools/lib/src/ios/migrations/uiscenedelegate_migration.dart

```diff
@@ -0,0 +1,94 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:meta/meta.dart' show visibleForTesting;
+import 'package:xml/xml.dart';
+
+import '../../base/file_system.dart';
+import '../../base/project_migrator.dart';
+import '../../xcode_project.dart';
+
+const String _manifestKey = 'UIApplicationSceneManifest';
+const String _storyboardKey = 'UIMainStoryboardFile';
+
+String _addition(String storyboard) => '''
+<key>UIApplicationSceneManifest</key>
+<dict>
+  <key>UIApplicationSupportsMultipleScenes</key>
+  <false/>
+  <key>UISceneConfigurations</key>
+  <dict>
+    <key>UIWindowSceneSessionRoleApplication</key>
+    <array>
+      <dict>
+        <key>UISceneClassName</key>
+        <string>UIWindowScene</string>
+        <key>UISceneDelegateClassName</key>
+        <string>FlutterSceneDelegate</string>
```

Comment:
should we also create a `SceneDelegate` that subclass from FlutterSceneDelegate, and use that instead? (it involves modifying xcodeproj since we are adding files, so not sure if it's feasible).

Reply:
I don't think so.  It just complicates the migration and adds zero benefit to users.

Reply:
I think we can leave it out, but just my 2 cents on the benefits: 

- It's more consistent with our existing template, where we generate `AppDelegate` that subclass from FlutterAppDelegate. 
- It's also more consistent with UIKit's template where Xcode creates both `AppDelegate` and `SceneDelegate` classes for developers to put logic in.  
- Although right now `FlutterSceneDelegate` is pretty much empty, we already get hints that `AppDelegate` is on its way out, so we will eventually move logic from `FlutterAppDelegate` into `FlutterSceneDelegate`. 

Counter arguments: 
- It complicates the migration (and I'm not sure if it's even doable)
- We can migrate after apple actually deprecates `AppDelegate`